integration_test/ops_agent_test.go

```diff
@@ -1957,84 +1957,77 @@ func TestPrometheusMetrics(t *testing.T) {
+			{local: path.Join(prometheusTestdata, "data.json"),
+				remote: path.Join("/opt", "go-http-server", "data.json")},
+			{local: path.Join(prometheusTestdata, "json_exporter_config.yaml"),
+				remote: path.Join("/opt", "json_exporter", "json_exporter_config.yaml")},
+			{local: path.Join(prometheusTestdata, "http-server-for-prometheus-test.service"),
+				remote: path.Join("/etc", "systemd", "system", "http-server-for-prometheus-test.service")},
+			{local: path.Join(prometheusTestdata, "json-exporter-for-prometheus-test.service"),
+				remote: path.Join("/etc", "systemd", "system", "json-exporter-for-prometheus-test.service")},
 		}
 
-		packages := []string{"wget", "python3"}
-		err := agents.InstallPackages(ctx, logger.ToMainLog(), vm, packages)
+		err := uploadFiles(ctx, logger, vm, testdataDir, filesToUpload)
 		if err != nil {
-			t.Fatalf("failed to install %v with err: %s", packages, err)
+			t.Fatal(err)
 		}
 
-		// Run the setup script to run the Python http server and the JSON exporter
-		setupScript, err := testdataDir.ReadFile(path.Join("testdata", "prometheus", "setup_json_exporter.sh"))
-		if err != nil {
-			t.Fatalf("failed to open setup script: %s", err)
+		if err := installGolang(ctx, logger, vm); err != nil {
+			t.Fatal(err)
 		}
 
-		env := map[string]string{"WORKDIR": workDir}
-		maxAttampts := 5
-		for attempt := 0; attempt < maxAttampts; attempt++ {
-			setupOut, err := gce.RunScriptRemotely(ctx, logger, vm, string(setupScript), nil, env)
-			// Since the script will start the processes in the background, the script should finish without error
-			if err != nil {
-				t.Fatalf("failed to run json exporter in VM with err: %v, stderr: %s", err, setupOut.Stderr)
-			}
-			// Wait until both are ready
-			time.Sleep(30 * time.Second)
-			liveCheckOut, liveCheckErr := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", `curl "http://localhost:7979/probe?module=default&target=http://localhost:8000/data.json"`)
-			// We will retry when:
-			// 1. JSON exporter is not started: in this case the stderr will have: "curl: (7) Failed to connect to localhost port 7979 after 1 ms: Connection refused"
-			// 2. The Python HTTP server is not started: in this case the stdout will not have the expected Prometheus style metrics
-			// If neither cases - break the retrying loop
-			if liveCheckErr == nil && !strings.Contains(liveCheckOut.Stderr, "Connection refused") && strings.Contains(liveCheckOut.Stdout, `test_counter_value{test_label="counter_label"} 1234`) {
-				break
-			}
-
-			// Out of attempts
-			if attempt == maxAttampts-1 {
-				errString := fmt.Sprintf("last stdout: %s last stderr: %s", liveCheckOut.Stdout, liveCheckOut.Stderr)
-				if liveCheckErr != nil {
-					errString = fmt.Sprintf("last err: %v %s", liveCheckErr, errString)
-				}
-				t.Fatalf("failed to start the HTTP server or JSON exporter - exhausted retries. %s", errString)
-			}
-		}
+		setupScript := `curl -L -o json_exporter.tar.gz \
```

Comment:
I see there was originally a script file for setting up the json exporter, and that it was removed in favour of this. What's the goal with that? I kind of like the script file more.

Reply:
Good question - so the script was doing two things: 
1. download the exporter, and 
2. run it and the http server in the background. 

2 was replaced by the systemd services because of better debug-ability: e.g., `systemctl restart json-exporter-for-prometheus-test` would immediately fail if the service, the binary or the exporter's config is not set up correctly in the VM, and logs of the services get uploaded as part of the test's output.  
For 1 and those `systemctl` commands - there is no particular reason to have them in there, and I agree having a separate script file would make the test cleaner - let me move them to a script file!